lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -460,9 +468,7 @@ def after_lazy(lazy_obj, owner:, field:, path:, scoped_context:, owner_object:,
         end
 
         def arguments(graphql_object, arg_owner, ast_node)
-          # Don't cache arguments if field extras or extensions are requested since they can mutate the argument data structure
-          if arg_owner.arguments_statically_coercible? &&
-              (!arg_owner.is_a?(GraphQL::Schema::Field) || (arg_owner.extras.empty? && arg_owner.extensions.empty?))
+          if arg_owner.arguments_statically_coercible? && !arg_owner.is_a?(GraphQL::Schema::Field)
```

Comment:
Should we remove the `!arg_owner.is_a?(GraphQL::Schema::Field)` check? I believe that was only needed to ensure we could call `#extras` and `#extensions`.

Reply:
Oh, I forgot. Let's see .... 

```suggestion
          if arg_owner.arguments_statically_coercible?
```